### PR TITLE
fix(spaces): resolve all space owners

### PR DESCRIPTION
**Pull request by andrewazores**
Correct behaviour of space owner resolution so that all owners are resolved, rather than only the
first owner across the result set

fixes https://github.com/fabric8-ui/ngx-fabric8-wit/issues/180

### CODE REVIEW

**Comment by christianvogt** [comment]
tests?

File: src/app/spaces/space.service.ts [hunk 1]
```diff
@@ -75,7 +75,7 @@ export class SpaceService {
         let newSpaces: Space[] = response.json().data as Space[];
         return newSpaces;
       })
-      .switchMap(spaces => {
+      .flatMap(spaces => {
```
Comment (joshuawilson) [inline src/app/spaces/space.service.ts]: please add the type for spaces

File: src/app/spaces/space.service.ts [hunk 2]
```diff
@@ -210,15 +210,16 @@ export class SpaceService {
         return Observable.empty<User>();
       }))
       // map the user objects back to the spaces to return a stream of spaces
-      .map(owner => {
-        if (owner) {
-          for (let space of spaces) {
-            space.relationalData = space.relationalData || {};
-            if (owner.id === space.relationships['owned-by'].data.id) {
+      .toArray()
+      .map(owners => {
```
Comment (joshuawilson) [inline src/app/spaces/space.service.ts]: please add type

**Comment by andrewazores** [comment]
tests added, code style cleaned up, rebased

**Pull request merged**